### PR TITLE
Separate `configuration` module

### PR DESCRIPTION
**Pull request by dmdashenkov**
We move the Protobuf types used for the validation gen configuration into a separate module. The module does not declare any dependencies on other Spine modules.
This will help us use this module in `mc-java` with no unnecessary dependency conflicts.

### CODE REVIEW

**Comment by dmdashenkov** [comment]
@armiol, PTAL.

**Review by armiol — APPROVED** [review]
@dmdashenkov LGTM with a comment.

File: model/src/main/java/io/spine/validation/RequiredIdPatternPolicy.java [hunk 1]
```diff
@@ -75,14 +76,16 @@
 
     private static boolean matches(FilePath path, FilePattern pattern) {
         String filePath = path.getValue();
-        switch (pattern.getValueCase()) {
+        FilePattern.KindCase kind = pattern.getKindCase();
+        checkNotNull(kind, "File pattern has unknown kind: %s.", pattern);
+        switch (kind) {
```
[the remark this answers is not in the document]
Comment (dmdashenkov) [reply model/src/main/java/io/spine/validation/RequiredIdPatternPolicy.java:81]: ignore

[the remark this answers is not in the document]
Comment (dmdashenkov) [reply model/src/main/java/io/spine/validation/RequiredIdPatternPolicy.java:81]: This is a false positive.

**Pull request merged**